tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.config.spring;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import org.mule.api.MuleException;
+import org.mule.api.lifecycle.Disposable;
+import org.mule.api.lifecycle.Startable;
+import org.mule.tck.junit4.FunctionalTestCase;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+public class SpringRegistryConcurrenctyTestCase extends FunctionalTestCase
+{
+
+    private static final int THREAD_COUNT = 10;
+    private static final long LATCH_TIMEOUT = 5;
+    private static final TimeUnit LATCH_TIME_UNIT = TimeUnit.SECONDS;
+
+    private CountDownLatch latch;
+    private List<Thread> threads;
+    private List<Map<String, Object>> reads;
+
+    @Override
+    protected String[] getConfigFiles()
+    {
+        return new String[] {};
```

Comment:
I need a started MuleContext with a backing SpringRegistry. This is the easiest way of achieving that. I could just use an AbstractMuleTestCase and initialise that on my own but then I'd be repeating code that FunctionalTestCase already has